From cb5cc122b4a5ab3db05ca3b0379278213d83039b Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Wed, 22 Apr 2015 20:56:29 +0000
Subject: [PATCH 1176/2748] HADOOP-11868. Invalid user logins trigger large
 backtraces in server log. Contributed by Chang Li
 (cherry picked from commit
 0ebe84d30af2046775884c9fb1e054da31582657)

(cherry picked from commit 2419861cac53c9bc9a14a635700f88d08c3105c4)

Change-Id: I233631a534c122b9db41a3d56579dbe51eaba4d4
---
 .../server/AuthenticationFilter.java               |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/AuthenticationFilter.java b/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/AuthenticationFilter.java
index 7e8a7bd..bf44f48 100644
--- a/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/AuthenticationFilter.java
+++ b/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/AuthenticationFilter.java
@@ -580,7 +580,11 @@ public Principal getUserPrincipal() {
       // exception from the filter itself is fatal
       errCode = HttpServletResponse.SC_FORBIDDEN;
       authenticationEx = ex;
-      LOG.warn("Authentication exception: " + ex.getMessage(), ex);
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Authentication exception: " + ex.getMessage(), ex);
+      } else {
+        LOG.warn("Authentication exception: " + ex.getMessage());
+      }
     }
     if (unauthorizedResponse) {
       if (!httpResponse.isCommitted()) {
-- 
1.7.9.5

